packages/synapse-react-client/src/components/TanStackTable/StyledTanStackTable.tsx

```diff
@@ -1,22 +1,56 @@
-import React from 'react'
-import { flexRender, Table } from '@tanstack/react-table'
+import React, { useMemo } from 'react'
+import { Cell, flexRender, Table } from '@tanstack/react-table'
 import {
   StyledTableContainer,
   StyledTableContainerProps,
 } from '../styled/StyledTableContainer'
+import TableBody from './TableBody'
+import {
+  getColumnSizeCssVariable,
+  getHeaderSizeCssVariable,
+} from './TanStackTableUtils'
 
 type StyledTanStackTableProps<T = unknown> = {
   table: Table<T>
   styledTableContainerProps?: StyledTableContainerProps
+  cellRenderer?: (cell: Cell<T, unknown>) => React.ReactNode
 }
 
 export default function StyledTanStackTable<T = unknown>(
   props: StyledTanStackTableProps<T>,
 ) {
-  const { table, styledTableContainerProps } = props
+  const { table, styledTableContainerProps, cellRenderer } = props
+
+  /**
+   * Instead of calling `column.getSize()` on every render for every header
+   * and especially every data cell (very expensive),
+   * we will calculate all column sizes at once at the root table level in a useMemo
+   * and pass the column sizes down as CSS variables to the <table> element.
+   *
+   * See https://tanstack.com/table/v8/docs/guide/column-sizing#advanced-column-resizing-performance
```

Comment:
Was able to lift the code from the example and made very few changes to adapt to use in this component.

Before, when columns were resized, every single cell had to be re-rendered by React to update the `width` property in the DOM.

Now, updating column sizes just updates these CSS variables which set the header/cell widths, and we use React.memo to only render the table body when the table data changes (not when the sizes change). So resizing a column works without React re-rendering every single table cell.